src/query/frontend/ast/cypher_main_visitor.cpp

```diff
@@ -1180,13 +1181,20 @@ antlrcpp::Any CypherMainVisitor::visitCallProcedure(MemgraphCypher::CallProcedur
   const auto &maybe_found =
       procedure::FindProcedure(procedure::gModuleRegistry, call_proc->procedure_name_, utils::NewDeleteResource());
   if (!maybe_found) {
-    throw SemanticException("There is no procedure named '{}'.", call_proc->procedure_name_);
+    const auto mg_specific_name = procedure::FindApocReplacement(call_proc->procedure_name_);
+    if (mg_specific_name) {
+      if (*mg_specific_name == "mgps.validate") {
```

Comment:
Should this come from the config file? If the implementation is more generic that this part here shouldn't exists in the cypher main visitor? All the logic should be done when the cursors are instantiated?

Reply:
The config file is for the mapping only. "mgps.validate" is a very special case. The issue is that so far this is the only known case where we have the 'no YIELD' related semantics to deal with the procedure, in fact we are not able to create such procedures with the current implementation. You are right that the logic should be as close to the cursor instantiation as possible, however this is the place where have to verify the semantics of the query.
It is always the case that if we call a procedure without yielding that is a semantic error, except in the case of  "mgps.validate", which is not even a procedure, it is a hardcoded cursor.

I do not think we should push semantic checks further down from here.
Once we have void procedures, this will not be an issue at all, we will be able to handle void procedures as actual query procedures instead of hardcoded, hacky cursor implementations.

IMO with our current possibilities this is out best bet. Yes, we can make this more generic, but that would be just propagating already flawed design down the line.

Let me know what you think.

Reply:
you can use and here instead of having multiple indentations, short circuit will solve problem

Reply:
You could also use a constant for the name